### ISSUE TITLE
power simulation takes a long time to complete

### ISSUE THREAD

**Issue by drajarshi** [accel-sim/accel-sim-framework#120]
Hi
I am trying to get an optimal configuration for a Quadro P2200 (sm_61). I started with the defined configuration of TITANX. After selecting the best configuration suggested by the tuner step, I added `power_simulation_enabled 1`. However, the simulation now hangs.

With `gdb`, I see the following :
```
(gdb) bt
#0  0x00007f94747773b2 in pipelined_simd_unit::get_active_lanes_in_pipeline (this=this@entry=0x5560b6ae53c0) at shader.cc:2263
#1  0x00007f9474777629 in int_unit::active_lanes_in_pipeline (this=0x5560b6ae53c0) at shader.cc:2299
#2  0x00007f9474779621 in shader_core_ctx::execute (this=this@entry=0x5560b410ffd0) at shader.cc:1747
#3  0x00007f947478d943 in shader_core_ctx::cycle (this=0x5560b410ffd0) at shader.cc:3498
#4  0x00007f947478d9d8 in simt_core_cluster::core_cycle (this=0x5560b410ff50) at shader.cc:4300
#5  0x00007f94747540d3 in gpgpu_sim::cycle (this=0x55609df85980) at gpu-sim.cc:1957
#6  0x000055609bfb5b76 in main (argc=<optimized out>, argv=<optimized out>) at main.cc:130
```
I print the stack after few seconds inside `gdb` and I see the same output.

Any suggestions?

**Comment by drajarshi** [comment]
The run eventually completed and took ~9 hours to complete. This is for a single kernel. 

The same kernel took ~33 mins to complete when run with the TITANX default tested configuration that's part of the source. Is this difference expected?

**Comment by VijayKandiah** [comment]
Hi,

This difference in simulation time is not expected. Enabling power simulation should not increase the simulation time by any significant amount. What kernel did you run?

**Comment by drajarshi** [comment]
Hi

Thanks for your response.

I ran the GICOV kernel that's part of leukocyte-3.1. 
Maybe I phrased it differently: I meant that the simulation hangs while using the tuner suggested config. The addition of the `power_simulation_enabled` flag was a step in the run, but not the reason for triggering the slowdown.

The run that took 33 mins with the TITANX default tested configuration and the one that took 9 hours with the tuner-suggested config, both have `power_simulation_enabled = 1`.

The difference is in the config which I am trying to figure out why it takes so long.

thanks
